pkg/health/heartbeat/heartbeat.go

```diff
@@ -241,17 +242,33 @@ func UpdateDeltaSnapshotLease(ctx context.Context, logger *logrus.Entry, prevDel
 	}
 
 	renewedLease := deltaSnapLease.DeepCopy()
-	renewedLease.Spec.HolderIdentity = &revisionNumber
 	renewedTime := time.Now()
 	renewedLease.Spec.RenewTime = &metav1.MicroTime{Time: renewedTime}
 
+	//Update revision number only if revisions in existing lease is lower
+	if len(prevDeltaSnapshots) > 0 {
```

Comment:
Added changes to set `HolderIdentity` of delta snapshot lease only if there are no previous delta snapshots and `HolderIdentity` is nil
PTAL

Reply:
thanks for addressing this issue, I saw you have updated the lease with `deltaSnapLease.Spec.HolderIdentity=0` when `len(prevDeltaSnapshots) == 0 && deltaSnapLease.Spec.HolderIdentity == nil` .... Can you also check that this will not cause any issue for compaction job calculating no of events.

Reply:
As can be seen from [this](https://github.com/gardener/etcd-druid/blob/0469cc9a8c1e50c5aee8b08fcc632b89b9278366/controllers/compaction_lease_controller.go#L137) point onwards in the `compaction-lease-controller`, currently the controller will return at this point.
However, if delta lease `HolderIdentity` is set to 0, it will then use that while calculating revision difference which will also be an accurate result

Reply:
make sense, thanks